bndtools.core/src/bndtools/model/repo/RepositoryBundleUtils.java

```diff
@@ -17,26 +19,55 @@ public static VersionedClause convertRepoBundle(RepositoryBundle bundle) {
 		return new VersionedClause(bundle.getBsn(), attribs);
 	}
 
+	/**
+	 * Converts a RepositoryBundleVersion into a version or version range. e.g.
+	 * version='latest' or version='[1.2.3,2.0.0)'.
+	 *
+	 * @param bundleVersion
+	 * @param phase
+	 * @return a version or version range.
+	 */
 	public static VersionedClause convertRepoBundleVersion(RepositoryBundleVersion bundleVersion,
 		DependencyPhase phase) {
 		Attrs attribs = new Attrs();
 		if (RepoUtils.isWorkspaceRepo(bundleVersion.getParentBundle()
 			.getRepo()))
 			attribs.put(Constants.VERSION_ATTRIBUTE, VERSION_LATEST);
 		else {
+
 			StringBuilder builder = new StringBuilder();
 
 			builder.append(bundleVersion.getVersion()
 				.getMajor());
 			builder.append('.')
 				.append(bundleVersion.getVersion()
 					.getMinor());
+
+			// TODO why is this check?
 			if (phase != DependencyPhase.Build)
```

Comment:
I'm glad to get some hints, since this is my first code PR in bndtools and I am new to the Eclipse GUI development.

Maybe you could help me with what this check means. I left it in place, and the version check also works with it. 
But not sure why it is there.

Reply:
@pkriens 
I found a commit by @njbartlett  

Commit 814c0946d33e170be8e17bf5a8458a07cb507573

> fix #681: Dragging version of dependency to -buildpath => use Major.minor segments only

<img width="801" alt="image" src="https://github.com/bndtools/bnd/assets/188422/ff467dfd-bd67-4a48-9b23-9529cf13b1a0">

I tried dragging an version-entry from the Repository browser directly in the `-buildpath` and can confirm that this only uses the Major.minor segment.

<img width="476" alt="image" src="https://github.com/bndtools/bnd/assets/188422/8ddc2f32-8ae7-4c66-b32e-40311fd721f6">

What is the reason for using only the **Major.minor segment** here?

In case it is not needed anymore, the check `if (phase != DependencyPhase.Build)` could be removed, which would make the code simpler (avoiding the StringBuilder and use the `bundleVersion.getVersion()` directly.

Thoughts?

Reply:
Perhaps the reasoning is that two bundles with the same major.minor should be identical, from a build perspective. Micro version number pertains to implementation details, not to the API - so the API should be identical between two bundles that have the same major.minor version. Just a thought.

Reply:
Thanks @kriegfrj , I adjusted the code to handle both cases more explicit as it was before and added some comments. So the -buildpath logic in bnd.bnd stays as is, based on your statement. The  other case (.bndrun Editor) now adds a range.